src/server.js

```diff
@@ -181,15 +183,23 @@ function start(opts) {
         item,
         id,
         opts.publicUrl,
-        (mbtiles, fromData) => {
+        (fileid, fromData) => {
           let dataItemId;
           for (const id of Object.keys(data)) {
             if (fromData) {
-              if (id === mbtiles) {
+              if (id === fileid) {
                 dataItemId = id;
               }
             } else {
-              if (data[id].mbtiles === mbtiles) {
+              if (
+                data[id].mbtiles !== undefined &&
```

Comment:
I'm not sure if this needs to check that `mbtiles` is both `!== undefined` and also `=== fileid`? Unless `fileid` might be undefined too?

Reply:
There are also these related pieces which has similar

In server_rendered.js there is this code to get the filename based on the id of what is specified in the style

https://github.com/maptiler/tileserver-gl/blob/a24a48bd1e550ef956eb7e3acb5572b2fb8ded82/src/serve_rendered.js#L1492-L1495

Then it uses the code at to get that filename and filetype
https://github.com/maptiler/tileserver-gl/blob/a24a48bd1e550ef956eb7e3acb5572b2fb8ded82/src/server.js#L242-L273

Passing the type back like this was somewhat to allow pmtiles to work with existing styles that use mbtiles:// designator for the source. In the style this allows you to put either mbtiles:// or pmtiles:// and it will still match based on the data id. it uses the type passed back to choose which source type to use.

Reply:
I think I've improved this. I not find the first item under data[id] and see if that matches fileid